### PR TITLE
[TorchTrain][Checkpoint] Update train state to include global_avg_losses and global_max_losses

### PR DESCRIPTION
**Pull request by wz337**
Based on discussion with @tianyu-l, we decided to only checkpoint `global_avg_losses` and `global_max_losses` per log frequency iteration to avoid all_reduce and device sync every iteration. `TrainState.current_loss` and `TrainState.losses` are removed from TrainState `state_dict()` and `load_state_dict()` call.

Tested with saving/loading with 30 steps with log_frequency = 10 and loading with 40 steps to resume training. The numerics in global_avg_losses and global_max_losses in the list aligns with expected.

```
Step 30 save:
[rank0]:before save: 
self.states['train_state']=TrainState(step=30, global_avg_losses=[10.8023841381073, 9.556332552433012, 6.9460043668746945], global_max_losses=[10.813274383544922, 10.74332332611084, 7.8649702072143555], log_steps=[1, 11, 21])


Step 30 load:
[rank0]:after load:
self.states['train_state']=TrainState(step=30, global_avg_losses=[10.8023841381073, 9.556332552433012, 6.9460043668746945], global_max_losses=[10.813274383544922, 10.74332332611084, 7.8649702072143555], log_steps=[1, 11, 21])


Step 40 load and resume training:
[rank0]:before save: 
self.states['train_state']=TrainState(step=40, global_avg_losses=[10.8023841381073, 9.556332552433012, 6.9460043668746945, 5.596909999847412], global_max_losses=[10.813274383544922, 10.74332332611084, 7.8649702072143555, 5.6796345710754395], log_steps=[1, 11, 21, 31])
```

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -51,20 +52,34 @@ class TrainState:
     step: int = 0
     current_loss: float = -1
     losses: List[float] = field(default_factory=list)
+    global_avg_losses: List[float] = field(default_factory=list)
+    global_max_losses: List[float] = field(default_factory=list)
     iter_times: List[float] = field(default_factory=list)
     data_load_times: List[float] = field(default_factory=list)
```
Comment (tianyu-l) [inline train.py]: Let's remove other unused variables (and their occurrences in this file) in `TrainState`, only keeping `step`, `global_avg_losses`, `global_max_losses` here.

Besides, we need an extra array (e.g. called `log_steps`) to save the steps at which we append to `global_avg_losses`, because after checkpoint loading, `log_freq` may have changed and this `log_steps` is needed in ploting the loss curve in TensorBoard.

File: train.py [hunk 2]
```diff
@@ -306,6 +325,9 @@ def main(job_config: JobConfig):
                 else:
                     global_avg_loss, global_max_loss = avg_loss, max_loss
```
Comment (tianyu-l) [inline train.py:328]: append `train_state.step` to `TrainState.log_steps` here
